pkgs/by-name/f2/f2/package.nix

```diff
@@ -4,23 +4,23 @@
   buildGoModule,
 }:
 
-buildGoModule rec {
+buildGoModule (finalAttrs: {
   pname = "f2";
-  version = "2.0.3";
+  version = "2.1.1";
 
   src = fetchFromGitHub {
     owner = "ayoisaiah";
     repo = "f2";
-    rev = "v${version}";
-    sha256 = "sha256-AjuWaSEP2X3URZBPD05laV32ms/pULooSQKXUz8sqsU=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-hl4giLTQtqJiPseiTzWPtksEYlyQpE1UOC7JMUF9v4Y=";
   };
 
-  vendorHash = "sha256-xKw9shfAtRjD0f4BGALM5VPjGOaYz1IqXWcctHcV/p8=";
+  vendorHash = "sha256-xeylGT32bGMJjGdpQQH8DBpqxtvMxpqSEsLPbeoUzl4=";
 
   ldflags = [
     "-s"
     "-w"
-    "-X=main.Version=${version}"
+    "-X=github.com/ayoisaiah/f2/v2/app.VersionString=${finalAttrs.version}"
   ];
 
   # has no tests
```

Comment:
Sorry, late to the party.

This package has tests now and some of them do not pass:
```
f2> --- FAIL: TestCustomSort (0.01s)
f2>     --- FAIL: TestCustomSort/sort_dng_files_by_time_variable (0.00s)
f2>         find_test.go:274: Failed to initialise exiftool: error when executing command: exec: "exiftool": executable file not found in $PATH
```

It would be good to enable tests and fix those failing ones.

Adding an update script is another potential improvement.

Reply:
Seems like it just needs some additional check inputs.

Reply:
Thanks for pointing that out @jian-lin. I'll make a PR soon.

Reply:
Let's continue in #408089.